docs/guides/developing/usage_examples.rst

```diff
@@ -236,45 +236,180 @@ Python
-   The user is required to call
-   :cpp:func:`LegacyViewingPipeline::setDisplayViewTransform` to set the
-   DisplayViewTransform.
-#. **Set any additional LegacyViewingPipeline options.**
-   If the user wants to specify a channel swizzle, a scene-linear exposure
-   offset, an artistic look, this is the place to add it. See ociodisplay for an
-   example. Note that although we provide recommendations for display, any
-   transforms are allowed to be added into any of the slots. So if for your app
-   you want to add 3 transforms into a particular slot (chained together), you
-   are free to wrap them in a :cpp:class:`GroupTransform` and set it
-   accordingly!
-#. **Get the processor from the LegacyViewingPipeline.**
-   The processor is then queried from the LegacyViewingPipeline.
-#. **Convert your image, using the processor.**
-   See :ref:`usage_applybasic` for details for using the CPU.
+.. code-block:: cpp
+
+   // Step 1: Get the config
+   OCIO::ConstConfigRcPtr config = OCIO::GetCurrentConfig();
+
+   // Step 2: Lookup the display ColorSpace
+   const char * device = config->getDefaultDisplayDeviceName();
+   const char * transformName = config->getDefaultDisplayTransformName(device);
+   const char * displayColorSpace = config->getDisplayColorSpaceName(device, transformName);
+
+   // Step 3: Create a DisplayTransform, and set the input and display ColorSpaces
+   // (This example assumes the input is scene linear. Adapt as needed.)
+
+   OCIO::DisplayTransformRcPtr transform = OCIO::DisplayTransform::Create();
+   transform->setInputColorSpaceName( OCIO::ROLE_SCENE_LINEAR );
+   transform->setDisplayColorSpaceName( displayColorSpace );
+
+   // Step 4: Add custom transforms for a 'canonical' Display Pipeline
+
+   // Add an fstop exposure control (in SCENE_LINEAR)
+   float gain = powf(2.0f, exposure_in_stops);
+   const float slope3f[] = { gain, gain, gain };
+   OCIO::CDLTransformRcPtr cc =  OCIO::CDLTransform::Create();
+   cc->setSlope(slope3f);
+   transform->setLinearCC(cc);
+
+   // Add a Channel view 'swizzle'
+
+   // 'channelHot' controls which channels are viewed.
+   int channelHot[4] = { 1, 1, 1, 1 };  // show rgb
+   //int channelHot[4] = { 1, 0, 0, 0 };  // show red
+   //int channelHot[4] = { 0, 0, 0, 1 };  // show alpha
+   //int channelHot[4] = { 1, 1, 1, 0 };  // show luma
+
+   float lumacoef[3];
+   config.getDefaultLumaCoefs(lumacoef);
+
+   float m44[16];
+   float offset[4];
+   OCIO::MatrixTransform::View(m44, offset, channelHot, lumacoef);
+   OCIO::MatrixTransformRcPtr swizzle = OCIO::MatrixTransform::Create();
+   swizzle->setValue(m44, offset);
+   transform->setChannelView(swizzle);
+
+   // And then process the image normally.
+   OCIO::ConstProcessorRcPtr processor = config->getProcessor(transform);
+
+   OCIO::PackedImageDesc img(imageData, w, h, 4);
+   processor->apply(img);
+
+Python
+++++++
+
+.. code-block:: python
+
+   def main():
```

Comment:
Please remove this line and the call to main() below.

Reply:
Hey @doug-walker, I removed def main(): and the main() call at the bottom but left ..code-block:: python so that the code is formatted as python code.